metricbeat/docs/metricbeat-in-a-container.asciidoc

```diff
@@ -0,0 +1,73 @@
+== Running Metricbeat in a Container
+
+Elastic does not provide any official container images for Metricbeat. The
+examples on this page assume you are using your own Metricbeat container image.
+
+Metricbeat can be executed in a container and when doing so there are some
+important things to be aware of if you want to monitor the host machine or other
+containers. Let's walk-through some examples using Docker as our container
+orchestration tool. This first example highlights the changes required to make
+the system module work properly.
+
+[source,sh]
+---
+sudo docker run \
+  --volume=/proc:/hostfs/proc:ro \ <1>
+  --volume=/sys/fs/cgroup:/hostfs/sys/fs/cgroup:ro \ <2>
+  --volume=/:/hostfs:ro \ <3>
+  --net=host <4>
+  my/metricbeat:latest -system.hostfs=/hostfs
+---
+
+<1> Metricbeat's system module collects much of its data through the Linux proc
+filesystem which is normally located at `/proc`. Due to the fact that containers
+are isolated as much as possible from the host, the data inside of the
+containers `/proc` is different than the host's `/proc`. To account for this you
+can mount the host's `/proc` filesystem inside of the container and tell
+Metricbeat to look inside the `/hostfs` directory when looking for `/proc` by
+using the `-system.hostfs=/hostfs` CLI flag.
+<2> If you have enabled cgroup reporting from the system process metricset then
+you need to mount the host's cgroup mountpoints within the container. They
+need to be mounted inside the directory specified by the `-system.hostfs` CLI
+flag.
+<3> If you want to be able to monitor filesystems from the host using the
+system filesystem metricset then those filesystems need to be mounted inside
+of the container. They can be mounted at any location.
+<4> The system network metricset uses data from `/proc/net/dev`, or
+`/hostfs/proc/net/dev` when using `-system.hostfs=/hostfs`, and the only way
+to make this file contain the host's network devices is to use the `--net=host`
+flag. This is due to Linux namespacing, simply bind mounting the host's `/proc`
+to `/hostfs/proc` is not sufficient.
+
+Next let's look at an example of monitoring a containerized service from a
```

Comment:
Perhaps add a titel here to make it clear, these are two separate topics. First monitor host system, second monitor service.